src/Diagnostics/FxCop/System.Runtime.Analyzers/CSharp/Globalization/UseOrdinalStringComparison.Fixer.cs

```diff
@@ -0,0 +1,84 @@
+                var newIdentifier = syntaxFactoryService.IdentifierName(newOrdinalText);
+                var newMemberAccess = memberAccess.WithName((SimpleNameSyntax)newIdentifier).WithAdditionalAnnotations(Formatter.Annotation);
+                var newRoot = root.ReplaceNode(memberAccess, newMemberAccess);
+                return Task.FromResult(document.WithSyntaxRoot(newRoot));
+            }
+
+            return Task.FromResult(document);
+        }
+
+        protected override bool IsInIdentifierNameContext(SyntaxNode node)
+        {
+            return node.IsKind(SyntaxKind.IdentifierName) &&
+                   node?.Parent?.FirstAncestorOrSelf<InvocationExpressionSyntax>() != null;
+        }
+
+        protected override async Task<Document> FixIdentifierName(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode identifier, CancellationToken cancellationToken)
+        {
+            var invokeParent = identifier?.Parent?.FirstAncestorOrSelf<InvocationExpressionSyntax>();
+            if (invokeParent != null)
+            {
+                var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var methodSymbol = model.GetSymbolInfo((IdentifierNameSyntax)identifier, cancellationToken).Symbol as IMethodSymbol;
+                if (methodSymbol != null && CanAddStringComparison(methodSymbol))
+                {
+                    // append a new StringComparison.Ordinal argument
+                    var newArg = syntaxFactoryService.Argument(CreateOrdinalMemberAccess(syntaxFactoryService, model))
+                        .WithAdditionalAnnotations(Formatter.Annotation);
+                    var newInvoke = invokeParent.AddArgumentListArguments((ArgumentSyntax)newArg).WithAdditionalAnnotations(Formatter.Annotation);
+                    var newRoot = root.ReplaceNode(invokeParent, newInvoke);
+                    return document.WithSyntaxRoot(newRoot);
+                }
+            }
+
+            return document;
+        }
+
+        protected override bool IsInEqualsContext(SyntaxNode node)
+        {
+            return node.IsKind(SyntaxKind.EqualsExpression) || node.IsKind(SyntaxKind.NotEqualsExpression);
+        }
+
+        protected override async Task<Document> FixEquals(Document document, SyntaxGenerator syntaxFactoryService, SyntaxNode root, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var binaryExpression = (BinaryExpressionSyntax)node;
+            var invocation = CreateEqualsExpression(syntaxFactoryService, model, binaryExpression.Left, binaryExpression.Right, node.Kind() == SyntaxKind.EqualsExpression).WithAdditionalAnnotations(Formatter.Annotation);
```

Comment:
Hmmm... Looks like SyntaxGenerator doesn't support replacing one type of expression with another? Is that why this code can't shared be in the base type?